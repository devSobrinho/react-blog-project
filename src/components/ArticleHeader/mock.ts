import { ArticleHeaderProps } from '.';
import mockArticleMeta from '../ArticleMeta/mock';
import mockHeader from '../Header/mock';

const header = {
  ...mockHeader,
  showText: false,
};

export default {
  header: header,
  title: 'Ipsum movet liberavisse an mea, diceret vituperata',
  excerpt:
    'Abhorreant reformidans eum an, per equidem repudiandae no, ex cum elitr argumentum persequeris.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  articleMeta: mockArticleMeta,
} as ArticleHeaderProps;
