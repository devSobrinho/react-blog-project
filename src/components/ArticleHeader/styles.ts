import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};
    padding-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    & ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      /* color: ${theme.colors.darkText}; */
      font-size: ${theme.font.sizes.huge};
    }

    @media${theme.media.lteMedium} {
      & ${HeadingStyles} {
        font-size: ${theme.font.sizes.xlarge};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
