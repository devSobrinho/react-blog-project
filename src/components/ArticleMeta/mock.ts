import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'daniel-sobrinho',
    displayName: 'Daniel Sobrinho',
  },
  categories: [
    {
      id: 'nnfgr3n434232',
      displayName: 'React',
      slug: 'react',
    },
    {
      id: 'a33d32d32fafa',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
