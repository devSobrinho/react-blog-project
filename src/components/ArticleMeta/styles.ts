import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    @media${theme.media.lteMedium} {
      p {
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
