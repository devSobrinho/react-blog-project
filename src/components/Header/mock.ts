import { HeaderProps } from '.';

export default {
  blogName: 'Daniel Sobrinho',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  showText: true,
  logo: {
    url: 'assets/images/dev-sobrinho.jpeg',
    id: 'qqqq11w1',
    alternativeText: 'textoalte',
  },
} as HeaderProps;
