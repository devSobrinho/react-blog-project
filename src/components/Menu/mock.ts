import { MenuProps } from '.';

export default {
  logoLink: {
    text: 'LogoLink',
    srcImg: 'assets/images/dev-sobrinho.jpeg',
    link: '/',
  },
  links: [
    { children: 'MenuLink1', link: '#home1', id: 'link-1', newTab: false },
    { children: 'MenuLink2aadsd', link: '#home2', id: 'link-2', newTab: false },
    { children: 'MenuLink3', link: '#home3', id: 'link-3', newTab: false },
    { children: 'MenuLink4asd', link: '#home4', id: 'link-4', newTab: false },
    {
      children: 'MenuLink5ssssssssssssssssssssssaaaa',
      link: '#home5',
      id: 'link-5',
      newTab: false,
    },
    { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLinkaaaaa', link: '#home6', id: 'link-6', newTab: false },
  ],
} as MenuProps;
