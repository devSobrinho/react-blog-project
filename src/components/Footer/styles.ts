import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
