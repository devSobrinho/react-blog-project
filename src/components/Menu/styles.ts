import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as MenuLinkStyles } from '../MenuLink/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

export const Wrapper = styled.menu`
  ${() => css`
    position: fixed;
    top: 0;
    z-index: 1;
  `}
`;

const menuVisibleMobile = () => css`
  padding: 3px;
  margin-left: calc(100vw - 39px - 1.6rem);
`;
const menuVisibleDesk = () => css`
  margin-left: 14rem;
`;

export const MenuOpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    border-radius: 5px;
    border-color: transparent;
    padding: 5px;
    margin-left: 1.6rem;
    margin-top: 1.6rem;
    transition: all 300ms ease-in-out;
    &:hover {
      color: ${theme.colors.secondary};
    }

    ${menuVisible && menuVisibleDesk()}

    @media ${theme.media.lteSmall} {
      ${menuVisible && menuVisibleMobile()}
    }
  `}
`;

export const Container = styled.nav<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    width: ${menuVisible ? '20rem' : '1rem'};
    transition: all 300ms ease-in-out;
    height: 100vh;
    background: ${theme.colors.primary};
    padding: ${menuVisible ? theme.spacings.medium : '0.5rem'};
    overflow-y: auto;
    overflow: auto;
    overflow-wrap: anywhere;

    & ${HeadingStyles} {
      width: auto;
      margin-top: 5rem;
    }

    & ${LogoLinkStyles} img {
      border: 0.5rem solid ${theme.colors.secondary};
      border-radius: 50%;
    }

    .menu-navbar {
      list-style: none;
      padding: 0;
      margin: 0;
      width: auto;
    }

    .navbar-item ${MenuLinkStyles} {
    }

    @media ${theme.media.lteSmall} {
      width: ${menuVisible ? '100vw' : '0px'};
      padding: ${menuVisible ? theme.spacings.medium : '0rem'};

      .menu-navbar {
        margin: 0 20px;
      }
      .navbar-item ${MenuLinkStyles} {
        width: initial;
        border: none;
      }
    }
  `}
`;
