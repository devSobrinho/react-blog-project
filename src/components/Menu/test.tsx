import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Menu, MenuProps } from '.';
import mock from './mock';

const props: MenuProps = mock;

describe('<Menu />', () => {
  it('should render and show menu button open', () => {
    renderTheme(<Menu {...props} />);

    expect(
      screen.getByRole('link', { name: 'Open or close menu' }),
    ).toHaveAttribute('href', '#');
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'menu' })).toHaveStyle({
      padding: '0.5rem',
    });
    expect(
      screen.getByRole('navigation', { name: 'menu' }).firstChild,
    ).not.toBeInTheDocument();
  });

  it('should open/close menu on button click ', () => {
    renderTheme(<Menu {...props} />);

    const buttonMenuClose = screen.getByRole('link', {
      name: 'Open or close menu',
    });
    fireEvent.click(buttonMenuClose);

    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'menu' })).toHaveStyle({
      padding: '2.4rem',
    });
    expect(screen.getByRole('img', { name: props.text })).toHaveAttribute(
      'src',
      props.srcImg,
    );
    expect(
      screen.getByRole('navigation', { name: 'menu' }).firstChild,
    ).toBeInTheDocument();
    expect(
      screen
        .getByRole('navigation', { name: 'menu' })
        .querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(6);
    expect(screen.getAllByRole('link', { name: 'MenuLink1' })[0]);

    fireEvent.click(buttonMenuClose);
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'menu' })).toHaveStyle({
      padding: '0.5rem',
    });
    expect(
      screen.getByRole('navigation', { name: 'menu' }).firstChild,
    ).not.toBeInTheDocument();
  });
  it('should render without img and snapshot', () => {
    const { container } = renderTheme(<Menu {...props} srcImg={undefined} />);

    const buttonMenuClose = screen.getByRole('link', {
      name: 'Open or close menu',
    });
    fireEvent.click(buttonMenuClose);

    expect(screen.getByRole('link', { name: 'LogoLink' })).toBeInTheDocument();
    expect(
      screen.queryByRole('img', { name: 'LogoLink' }),
    ).not.toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Menu {...props} />);

    expect(container.firstChild).toMatchSnapshot();

    const buttonMenuClose = screen.getByRole('link', {
      name: 'Open or close menu',
    });
    fireEvent.click(buttonMenuClose);
    expect(container.firstChild).toMatchSnapshot();
  });
});
