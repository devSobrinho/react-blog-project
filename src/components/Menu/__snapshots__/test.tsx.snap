// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
.c0 {
  position: fixed;
  top: 0;
  z-index: 1;
}

.c1 {
  position: fixed;
  color: #FFFFFF;
  background: #000000;
  border-radius: 5px;
  border-color: transparent;
  padding: 5px;
  margin-left: 1.6rem;
  margin-top: 1.6rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover {
  color: #dc143c;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  height: 100vh;
  background: #000000;
  padding: 0.5rem;
  overflow-y: auto;
  overflow: auto;
  overflow-wrap: anywhere;
}

.c3 .menu-navbar {
  list-style: none;
  padding: 0;
  margin: 0;
  width: auto;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:425px) {

}

@media (max-width:425px) {
  .c3 {
    width: 0px;
    padding: 0rem;
  }

  .c3 .menu-navbar {
    margin: 0 20px;
  }
}

<menu
  class="c0"
>
  <a
    aria-label="Open or close menu"
    class="c1 menu-icon"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="2.5rem"
      viewBox="0 0 24 24"
      width="2.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <nav
    aria-label="menu"
    class="c3 menu"
  />
</menu>
`;

exports[`<Menu /> should match snapshot 2`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
}

.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c9 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all,0.3s ease-in-out;
  transition: all,0.3s ease-in-out;
}

.c9:hover {
  color: #dc143c;
  border-right: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 0;
  z-index: 1;
}

.c1 {
  position: fixed;
  color: #FFFFFF;
  background: #000000;
  border-radius: 5px;
  border-color: transparent;
  padding: 5px;
  margin-left: 1.6rem;
  margin-top: 1.6rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  margin-left: 14rem;
}

.c1:hover {
  color: #dc143c;
}

.c10 .c4 {
  width: auto;
  margin-top: 5rem;
}

.c10 .c6 img {
  border: 0.5rem solid #dc143c;
  border-radius: 50%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  height: 100vh;
  background: #000000;
  padding: 2.4rem;
  overflow-y: auto;
  overflow: auto;
  overflow-wrap: anywhere;
}

.c3 .c4 {
  width: auto;
  margin-top: 5rem;
}

.c3 .c6 img {
  border: 0.5rem solid #dc143c;
  border-radius: 50%;
}

.c3 .menu-navbar {
  list-style: none;
  padding: 0;
  margin: 0;
  width: auto;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:425px) {

}

@media (max-width:425px) {
  .c1 {
    padding: 3px;
    margin-left: calc(100vw - 39px - 1.6rem);
  }
}

@media (max-width:425px) {
  .c10 .navbar-item .c8 {
    width: initial;
    border: none;
  }
}

@media (max-width:425px) {
  .c3 {
    width: 100vw;
    padding: 2.4rem;
  }

  .c3 .menu-navbar {
    margin: 0 20px;
  }

  .c3 .navbar-item .c8 {
    width: initial;
    border: none;
  }
}

<menu
  class="c0"
>
  <a
    aria-label="Open or close menu"
    class="c1 menu-icon-opened"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Close menu"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="2.5rem"
      viewBox="0 0 24 24"
      width="2.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
      />
    </svg>
  </a>
  <nav
    aria-label="menu"
    class="c3 menu"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="/"
        target="_self"
      >
        <img
          alt="LogoLink"
          src="assets/images/dev-sobrinho.jpeg"
        />
      </a>
    </h1>
    <ul
      class="menu-navbar"
    >
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home1"
          target="_self"
        >
          MenuLink1
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home2"
          target="_self"
        >
          MenuLink2aadsd
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home3"
          target="_self"
        >
          MenuLink3
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home4"
          target="_self"
        >
          MenuLink4asd
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home5"
          target="_self"
        >
          MenuLink5ss
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home6"
          target="_self"
        >
          MenuLink6
        </a>
      </li>
    </ul>
  </nav>
</menu>
`;

exports[`<Menu /> should render without img and snapshot 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
}

.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c9 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all,0.3s ease-in-out;
  transition: all,0.3s ease-in-out;
}

.c9:hover {
  color: #dc143c;
  border-right: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 0;
  z-index: 1;
}

.c1 {
  position: fixed;
  color: #FFFFFF;
  background: #000000;
  border-radius: 5px;
  border-color: transparent;
  padding: 5px;
  margin-left: 1.6rem;
  margin-top: 1.6rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  margin-left: 14rem;
}

.c1:hover {
  color: #dc143c;
}

.c10 .c4 {
  width: auto;
  margin-top: 5rem;
}

.c10 .c6 img {
  border: 0.5rem solid #dc143c;
  border-radius: 50%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  height: 100vh;
  background: #000000;
  padding: 2.4rem;
  overflow-y: auto;
  overflow: auto;
  overflow-wrap: anywhere;
}

.c3 .c4 {
  width: auto;
  margin-top: 5rem;
}

.c3 .c6 img {
  border: 0.5rem solid #dc143c;
  border-radius: 50%;
}

.c3 .menu-navbar {
  list-style: none;
  padding: 0;
  margin: 0;
  width: auto;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:425px) {

}

@media (max-width:425px) {
  .c1 {
    padding: 3px;
    margin-left: calc(100vw - 39px - 1.6rem);
  }
}

@media (max-width:425px) {
  .c10 .navbar-item .c8 {
    width: initial;
    border: none;
  }
}

@media (max-width:425px) {
  .c3 {
    width: 100vw;
    padding: 2.4rem;
  }

  .c3 .menu-navbar {
    margin: 0 20px;
  }

  .c3 .navbar-item .c8 {
    width: initial;
    border: none;
  }
}

<menu
  class="c0"
>
  <a
    aria-label="Open or close menu"
    class="c1 menu-icon-opened"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Close menu"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="2.5rem"
      viewBox="0 0 24 24"
      width="2.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
      />
    </svg>
  </a>
  <nav
    aria-label="menu"
    class="c3 menu"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="/"
        target="_self"
      >
        LogoLink
      </a>
    </h1>
    <ul
      class="menu-navbar"
    >
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home1"
          target="_self"
        >
          MenuLink1
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home2"
          target="_self"
        >
          MenuLink2aadsd
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home3"
          target="_self"
        >
          MenuLink3
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home4"
          target="_self"
        >
          MenuLink4asd
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home5"
          target="_self"
        >
          MenuLink5ss
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          class="c8 c9"
          href="#home6"
          target="_self"
        >
          MenuLink6
        </a>
      </li>
    </ul>
  </nav>
</menu>
`;
